presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/CommandExecutor.java

```diff
@@ -61,22 +67,75 @@ private int execute()
         processBuilder.redirectErrorStream(true);
         try {
             Process process = processBuilder.start();
-            new Thread(() -> {
-                try (InputStream processStream = process.getInputStream()) {
-                    LOGGER.info("Output from command: {}\n{}", commandString, new String(ByteStreams.toByteArray(processStream)));
-                }
-                catch (IOException e) {
-                    LOGGER.error("Failed to log the process output", e);
-                }
-            }).start();
-            if (!process.waitFor(timeout, TimeUnit.SECONDS)) {
+            CommandResult commandResult = new CommandResult();
+            ExecutorService executor = newFixedThreadPool(1);
+            Future<String> outputThread = executor.submit(new ExecuteCallable(process));
+            try {
+                commandResult.setOutput(outputThread.get(timeout, SECONDS));
+            }
+            catch (TimeoutException e) {
                 process.destroyForcibly();
                 throw new PrestoManagerException(format("Command timed out: %s", commandString));
             }
-            return process.exitValue();
+            finally {
+                executor.shutdownNow();
+            }
+            LOGGER.info("Output from command: {}\n{}", commandString, commandResult.getOutput());
+            commandResult.setExitValue(process.exitValue());
+            return commandResult;
         }
-        catch (IOException | InterruptedException e) {
+        catch (IOException | ExecutionException | InterruptedException | IllegalThreadStateException e) {
             throw new PrestoManagerException(format("Error executing command: %s", commandString), e);
         }
     }
+
+    public class ExecuteCallable
```

Comment:
This class can be replaced with this method, which has the exact body of `ExecuteCallable#call()`:
```java
private getProcessOutput(Process process) { ... }
```
Then you can replace the single instantiation of `ExecuteCallable` with a lambda:
```java
executor.submit(() -> getProcessOutput(process))
```